cmd/agent/common/autoconfig.go

```diff
@@ -107,3 +109,19 @@ func SetupAutoConfig(confdPath string) {
 func StartAutoConfig() {
 	AC.LoadAndRun()
 }
+
+// BlockUntilAutoConfigRanOnce blocks until the AutoConfig has been ran once.
+// It also returns after the given timeout.
+func BlockUntilAutoConfigRanOnce(timeout time.Duration) {
+	now := time.Now()
+	for {
+		time.Sleep(100 * time.Millisecond) // don't hog the CPU
+		if AC.HasRanOnce() {
+			return
+		}
+		if time.Since(now) > timeout {
+			log.Warn("BlockUntilAutoConfigRanOnce timeout after", timeout)
+			return
+		}
+	}
+}
```

Comment:
It looks like the goal of this function is to wait for a condition to be met (`AC.ranOnce` to be set to 1).
May be [`sync.Cond`](https://golang.org/pkg/sync/#Cond) could advantageously replace the use of `time.Sleep`.

[`sync.Cond`](https://golang.org/pkg/sync/#Cond) is specifically crafted to make a goroutine sleep until a given condition is met.

Reply:
Good point! I'll do the change.

Reply:
Actually using `sync.Cond` would be difficult for some reasons:
  - the part waiting for the block will still have to loop forever until the AC is instanciated and its `Cond` is instanciated and ready as well.
  - sharing the `sync.Cond` in this case is kinda painful because of some global variables of the AC.
Do you mind if I keep this implementation or would you really prefer for me to push forward with `sync.Cond`? I think the trade-off of this sleep brings some simplicity in the code + the capacity to timeout, which I would not able to do with a `sync.Cond`.